### PR TITLE
Lambda methods

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: vars/deployToLambda.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
+def call(Map config) {
+    library("tdr-jenkinslib")
+    pipeline {
+        agent {
+            label "master"
+        }
+        parameters {
+            choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are building the front end for")
```
Comment (TomJKing) [inline vars/deployToLambda.groovy]: Probably want to change the description for the STAGE parameter

File: vars/deployToLambda.groovy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+        post {
+            success {
+                script {
+                    tdr.runEndToEndTests(0, config.stage, BUILD_URL)
+                }
+            }
+        }
+    }
+}
```
Comment (TomJKing) [inline vars/deployToLambda.groovy]: Missing line break at end of file

File: vars/lambdaTests.groovy [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    def versionTag = "v${env.BUILD_NUMBER}"
+    def repo = "tdr-${config.libraryName}"
+
+    pipeline {
+        agent {
+            label "master"
+        }
+        parameters {
+            choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are building the front end for")
```
Comment (TomJKing) [inline vars/lambdaTests.groovy]: Probably want to change the description for the STAGE parameter

File: vars/lambdaTests.groovy [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                }
+            }
+            success {
+                script {
+                    tdr.reportSuccessfulBuildToGitHub(repo)
+                }
+            }
+        }
+    }
+
```
Comment (TomJKing) [inline vars/lambdaTests.groovy]: Extra line break

File: vars/lambdaTests.groovy [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                            script {
+                                unstash "${config.libraryName}-jar"
+                                tdr.copyToS3CodeBucket(config.libraryName, versionTag)
+
+                                tdr.configureJenkinsGitUser()
+
+                                sh "git tag ${versionTag}"
+                                sh "git checkout ${env.BRANCH_NAME}"
+
+                                tdr.pushGitHubBranch("master")
```
Comment (TomJKing) [inline vars/lambdaTests.groovy:53]: Didn't think we can push to the master branch with the branch protection rules in place?

File: vars/tdr.groovy [hunk 6]
```diff
@@ -33,6 +33,16 @@ def reportFailedBuildToGitHub(String repo) {
   }
 }
 
+def assembleAndStash(String libraryName) {
+  sh "sbt -no-colors assembly"
+  stash includes: "target/scala-2.13/${libraryName}.jar", name: "${libraryName}-jar"
```
Comment (TomJKing) [inline vars/tdr.groovy]: Is it worth adding "scala-2.13" as a "constant" to the library as its used in multiple places. Also would make it easier if the scala version is ever updated.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments and questions.

File: vars/deployToLambda.groovy (hunk 1 above)
Comment (MancunianSam) [reply vars/deployToLambda.groovy]: Good point, thanks

File: vars/deployToLambda.groovy (hunk 2 above)
Comment (MancunianSam) [reply vars/deployToLambda.groovy]: :smile:

File: vars/lambdaTests.groovy (hunk 3 above)
Comment (MancunianSam) [reply vars/lambdaTests.groovy]: I've changed this one as well.

File: vars/lambdaTests.groovy (hunk 4 above)
Comment (MancunianSam) [reply vars/lambdaTests.groovy]: I've removed it.

File: vars/lambdaTests.groovy (hunk 5 above)
Comment (MancunianSam) [reply vars/lambdaTests.groovy:53]: It seems to work if it's just a tag but you can't commit any file changes.

Comment (TomJKing) [reply vars/lambdaTests.groovy:53]: Fair enough

File: vars/tdr.groovy (hunk 6 above)
Comment (MancunianSam) [reply vars/tdr.groovy]: Makes sense, I've added a field at the top for the scala version. I've tried it by replaying an existing job with this in and it seems to work.

**Review by TomJKing — APPROVED** [review]
Looks good

**Pull request merged**